src/main/java/com/google/codeu/servlets/LocationServlet.java

```diff
@@ -69,30 +78,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         return;
 	}
     
-    List<Location> locations = datastore.getLocations(user);
+    Location userLocation = datastore.getUserLocation(user);
   
     
     /*check if location object exists*/
-    if(locations==null || locations.isEmpty())
+    if(userLocation==null)
     {
-      /*make location object*/
+      /*make a new location for user*/
       String text = "Charity Location";
-      Location location = new Location(latitude, longitude, text, user);
-      datastore.storeLocation(location);
+      userLocation = new Location(latitude, longitude, text, user);
+      datastore.storeLocation(userLocation);
     }
     else //location already exists => update it
     {
-      String existingLocationId = locations.get(0).getId().toString();
+      String existingLocationId = userLocation.getId().toString();
       datastore.updateLocation(existingLocationId, latitude, longitude); 
     }
     
-    locations = datastore.getLocations(user);
-     if(locations!=null && locations.size()>0) {
-      /*send notification to other charities near by*/
-      for(int i = 1; i<locations.size(); i++)
-      {
-        Notification notification = new Notification(user, locations.get(i).getUser(), "New charity location found near you!");//in this case link is also user: test@example.com
-        datastore.storeNotification(notification);
+    /*send notification to other users nearby, if the updated location is for a charity type user*/
+    User locationUser = datastore.getUser(user);
+    if(locationUser!=null && locationUser.getType()!=null && locationUser.getType()== User.CHARITY_TYPE) {
```

Comment:
I see that this set of conditions is used in a number of places. I suggest creating a new public final class (LocationUtil, which a private constructor to emphasize that it should not be instantiated and final to prevent it from being inherited from) and moving this into a public static boolean method and calling it throughout. This would would ensure that the conditions are the same everywhere and that if the conditions change, you'll only have to update single location vs accidentally missing one.

Reply:
Thank you so much, that's a great idea!